src/utils/load-mst-settings.ts

```diff
@@ -0,0 +1,230 @@
+import { applySnapshot, getSnapshot, getType, IAnyModelType, IAnyStateTreeNode, IAnyType, isArrayType, isFrozenType, isLiteralType, isModelType, isOptionalType, isUnionType, types } from "mobx-state-tree";
+import merge from "deepmerge";
+
+export type SettingsSourceValue = string | null | undefined;
+
+export abstract class SettingsSource {
+  abstract getItem(key: string): SettingsSourceValue;
+
+  abstract hasKeyStartingWith(keyPrefix: string): boolean;
+
+  getNumber(key: string, value: SettingsSourceValue): number | undefined {
+    // If the value is null or undefined we treat it as undefined
+    // This way settings sources like localStorage that return null for missing keys
+    // will not override existing values
+    // Number(null) is 0 which is not what we want
+    if (value == null) {
+      return undefined;
+    }
+    const num = Number(value);
+    return isNaN(num) ? undefined : num;
+  }
+
+  getBoolean(key: string, value: SettingsSourceValue): boolean | undefined {
+    // If the value is null or undefined we treat it as undefined
+    // This way settings sources like localStorage that return null for missing keys
+    // will not override existing values
+    // The URL Param source modifies this behavior so a parameter that exists with no value
+    // is treated as true
+    if (value == null) {
+      return undefined;
+    }
```

Comment:
Would it make sense to have a helper function named something like `isMissingSettingValue` for this and the similar code in `getNumber`? Even though the check is very simple, I'm thinking that would help reduce the redundant comments. Definitely not necessary, though, if you think having the two separate comments with the small differences is good.